src/components/computing-status/use-computing-status.ts

```diff
@@ -0,0 +1,151 @@
+
+interface UseComputingStatusProps {
+    (
+        studyUuid: UUID,
+        nodeUuid: UUID,
+        fetcher: (studyUuid: UUID, nodeUuid: UUID) => Promise<string>,
+        invalidations: string[],
+        resultConversion: (x: string) => RunningStatus,
+        computingType: ComputingType
+    ): void;
+}
+
+interface LastUpdateProps {
+    studyUpdatedForce: StudyUpdated;
+    fetcher: (studyUuid: UUID, nodeUuid: UUID) => Promise<string>;
+}
+
+function isWorthUpdate(
+    studyUpdatedForce: StudyUpdated,
+    fetcher: (studyUuid: UUID, nodeUuid: UUID) => Promise<string>,
+    lastUpdateRef: RefObject<LastUpdateProps>,
+    nodeUuidRef: RefObject<UUID>,
+    nodeUuid: UUID,
+    invalidations: Array<string>
+) {
+    const headers = studyUpdatedForce?.eventData?.headers;
+    const updateType = headers?.[UPDATE_TYPE_HEADER];
+    const node = headers?.['node'];
+    const nodes = headers?.['nodes'];
+    if (nodeUuidRef.current !== nodeUuid) {
+        return true;
+    }
+    if (fetcher && lastUpdateRef.current?.fetcher !== fetcher) {
+        return true;
+    }
+    if (
+        studyUpdatedForce &&
+        lastUpdateRef.current?.studyUpdatedForce === studyUpdatedForce
+    ) {
+        return false;
+    }
+    if (!updateType) {
+        return false;
+    }
+    if (invalidations.indexOf(updateType) <= -1) {
+        return false;
+    }
+    if (node === undefined && nodes === undefined) {
+        return true;
+    }
+    if (node === nodeUuid || nodes?.indexOf(nodeUuid) !== -1) {
+        return true;
+    }
+
+    return false;
+}
+
+// this hook loads <computingType> state into redux, then keeps it updated according to notifications
+/**
+ *
+ * @param studyUuid current study uuid
+ * @param nodeUuid current node uuid
+ * @param fetcher method fetching current <computingType> state
+ * @param invalidations when receiving notifications, if updateType is included in <invalidations>, this hook will update
+ * @param resultConversion converts <fetcher> result to RunningStatus
+ * @param computingType ComputingType targeted by this hook
+ */
+export const useComputingStatus: UseComputingStatusProps = (
+    studyUuid,
+    nodeUuid,
+    fetcher,
+    invalidations,
+    resultConversion,
+    computingType
+) => {
+    const nodeUuidRef = useRef<UUID | null>(null);
+    const studyUpdatedForce = useSelector(
+        (state: ReduxState) => state.studyUpdated
+    );
+    const lastUpdateRef = useRef<LastUpdateProps | null>(null);
+    const dispatch = useDispatch();
+
+    const update = useCallback(() => {
+        // this is used to prevent race conditions from happening
+        // if another request is sent, the previous one won't do anything
+        let canceledRequest = false;
+
+        nodeUuidRef.current = nodeUuid;
+        fetcher(studyUuid, nodeUuid)
+            .then((res: string) => {
+                if (!canceledRequest && nodeUuidRef.current === nodeUuid) {
```

Comment:
```suggestion
                if (!canceledRequest) {
```
Do we need it now that we prevent race condition ? (for me no, except if I missed something)

Reply:
since nodeUuidRef is historically used somewhere else in the hook, I prefer to keep it that way for now, the fix being already important enough